source/code/plugins/in_dsc_monitor.rb

```diff
@@ -85,6 +86,7 @@ def run_check_status
           @dsc_cache[:status] = dsc_status
           @dsc_cache.commit
         }
+        File.chmod(0644, dsc_cache_file)
```

Comment:
My worry is with that approach is YAML::Store changing the permissions when it runs over the file.